### PR TITLE
Adds NM Olgoi Khurkoi respawn timer

### PR DESCRIPTION
**Pull request by Aeltorian**
Adds a spawntimer on despawn to NM Olgoi Khurkoi in North Gustaberg [s]

**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Adds a respawn timer to Olgoi-K by setting a respawn timer when he despawns. Currently, set to a random math range between 90 to 120 minutes. 

References for respawn rates:

https://wikiwiki.jp/ffxi/Verup/091110/NM/%E3%82%A2%E3%83%AB%E3%82%BF%E3%83%8A%E3%82%A8%E3%83%AA%E3%82%A2#y3efabea

https://ffxiclopedia.fandom.com/wiki/Olgoi-Khorkhoi

## Steps to test these changes

Find Olgoi-Khurkoi out in North Gustaberg (S). Fight it and then upon death, wait anywhere between 90 to 120 minutes in order for it to respawn.

### CODE REVIEW

File: scripts/zones/North_Gustaberg_[S]/mobs/Olgoi-Khorkhoi.lua [hunk 1]
```diff
@@ -6,8 +6,17 @@ require("scripts/globals/hunts")
 -----------------------------------
 local entity = {}
 
+entity.onMobSpawn = function(mob)
+    mob:setRespawnTime(0, true)
```
Comment (claywar) [inline scripts/zones/North_Gustaberg_[S]/mobs/Olgoi-Khorkhoi.lua]: setRespawnTime only accepts one argument

Comment (Aeltorian) [reply scripts/zones/North_Gustaberg_[S]/mobs/Olgoi-Khorkhoi.lua]: So the true doesn't need to be there. Alright.

File: scripts/zones/North_Gustaberg_[S]/mobs/Olgoi-Khorkhoi.lua [hunk 2]
```diff
@@ -6,8 +6,17 @@ require("scripts/globals/hunts")
 -----------------------------------
 local entity = {}
 
+entity.onMobSpawn = function(mob)
+    mob:setRespawnTime(0)
```
Comment (zach2good) [inline scripts/zones/North_Gustaberg_[S]/mobs/Olgoi-Khorkhoi.lua]: Looking around the codebase, I don't think setting the respawn time is required - I think it's implied?

Possibly depending on what the flags in the db say for the mob

Comment (Aeltorian) [reply scripts/zones/North_Gustaberg_[S]/mobs/Olgoi-Khorkhoi.lua]: Just for the onMobSpawn since despawn is calling it?

**Review by zach2good — APPROVED** [review]
Note to reviewers: remember to squash!

**Pull request merged**